src/lib/BarStore.js

```diff
@@ -12,31 +12,119 @@ const CATEGORIES = ['cozy', 'partners', 'ptnb']
+    this.helpLink = ''
   }
 
-  getClaudyActions () {
-    if (this.claudyActions) return Promise.resolve(this.claudyActions)
-    return stack.get.context()
-      .then(context => {
-        const contextActions = (context.data && context.data.attributes && context.data.attributes['claudy_actions']) || null
-        if (!contextActions) return null
-        // get an arrays of action
-        const claudyActions = contextActions.map(slug => {
-          if (CLAUDY_ACTIONS.hasOwnProperty(slug)) {
-            // adding also the action slug
-            return Object.assign({}, CLAUDY_ACTIONS[slug], { slug })
+  async fetchApps () {
+    let apps
+    try {
+      apps = await Promise.all((await stack.get.apps())
+      .filter(app => !EXCLUDES.includes(app.attributes.slug))
+      .map(async app => {
+        const oldApp = this.installedApps.find(item => item.slug === app.attributes.slug)
+        let icon
+
+        if (oldApp && oldApp.icon.cached) {
+          icon = oldApp.icon
+        } else {
+          icon = {
+            src: await stack.get.icon(app.links.icon),
+            cached: true
           }
-        }).filter(action => action)
-        this.claudyActions = claudyActions
+        }
+
         return {
-          actions: claudyActions
+          editor: app.attributes.editor,
+          name: app.attributes.name,
+          slug: app.attributes.slug,
+          href: app.links.related,
+          category: CATEGORIES.includes(app.attributes.category) ? app.attributes.category : 'others',
+          icon
         }
+      }))
+      this.installedApps = apps
+    } catch (e) {
+      apps = [{error: e}]
+    }
+    return apps
+  }
+
+  fetchComingSoonApps () {
+    return stack.get.context()
+    .then(context => {
+      const comingSoonApps = (context.data && context.data.attributes &&
+      context.data.attributes['coming_soon'] &&
+      Object.values(context.data.attributes['coming_soon'])) || []
+
+      return comingSoonApps.map(app => {
+        let icon
+
+        try {
+          icon = app.slug && {
+            cached: true,
+            src: require(`../assets/icons/comingsoon/icon-${app.slug}.svg`)
+          }
+        } catch (error) {
+          console.warn && console.warn(`Cannot retrieve icon for app ${app.name}:`, error.message)
+        }
+
+        return Object.assign({}, app, {
+          comingSoon: true,
+          icon: icon
+        })
       })
-      .catch(error => {
-        console.warn && console.warn(`Cozy-bar cannot fetch Claudy: ${error.message}`)
-        return null
+    })
+    .catch(error => {
+      console.warn && console.warn(`Cozy-bar cannot fetch comming soon apps: ${error.message}`)
+      return []
+    })
+  }
+
+  async getAppsList () {
+    const apps = await this.fetchApps()
+    const comingSoonApps = await this.fetchComingSoonApps()
+    return apps.concat(comingSoonApps)
+  }
+
+  getClaudyActions () {
```

Comment:
You mix `async/await` syntax and `Promise.then` syntax. It is a little bit hard to read heterogeneous syntax. If there is no special unknown needs, could we homogenize them?